components/server/src/user/user-service.ts

```diff
@@ -77,12 +79,19 @@ export class UserService {
     }
 
     async findUserById(userId: string, id: string): Promise<User> {
-        await this.authorizer.checkPermissionOnUser(userId, "read_info", id);
+        if (userId !== id) {
+            await this.authorizer.checkPermissionOnUser(userId, "read_info", id);
+        }
         const result = await this.userDb.findUserById(id);
         if (!result) {
             throw new ApplicationError(ErrorCodes.NOT_FOUND, "not found");
         }
-        return result;
+        try {
+            return await this.relationshipUpdater.migrate(result);
```

Comment:
do we want to block for now? I remember you mentioned to run it in parallel to learn about performance, although we will need some metrics then?

Reply:
I see it is a feature flag, I'm still not sure how we are going to measure the perf impact, i.e. if we enable for Gitpod team only, how we can see perf changes only for such users?

Reply:
Yes, I decided to go with blocking since we have a feature flag. I should add metrics here as well.

Reply:
BTW I'm currently looking into adding `user -> workspace.owner`. This change might change the timing quite a bit, especially for people with prebuilds. But we can observe when testing with our own accounts. :+1: